Sources/Player/PlaybackEngine/Internal/Events/PlayerEventSender.swift

```diff
@@ -209,11 +209,17 @@ private extension PlayerEventSender {
 				print("StreamingPrivilegesHandler failed to get credentials")
 				return
 			}
-
+			
+			let userClientId: Int? = if let userClientIdSupplier {
+				userClientIdSupplier()
+			} else {
+				nil
+			}
+		
 			let user = User(
-				id: userConfiguration!.userId,
+				id: Int(userId!) ?? -1,
```

Comment:
This causes a crash if `userId` is nil

Reply:
Yes I was aware of that. If you look up the code right before, we do return if the user is not authorized. So it should not happen. And if it does, then we have a deeper issue that we should solve. So I decided to force a crash in this case. Does that make sense? Or do you still prefer we set a fallback here?

Reply:
yes, makes sense 👍 

Before we were also forcing a crash if we had an invalid userConfiguration so this is the same.